### PR TITLE
Toolbar refactoring, unification; search for contacts (#448); processed-messages cache (#552); fix for #554

### PR DESCRIPTION
**Pull request by alwx**

### CODE REVIEW

**Comment by alwx** [comment]
This PR contains several things:
1) It fixes #448 and adds the ability to search and remove contacts;
2) It makes toolbar with search field more general — now we use the same search field everywhere and it behaves and looks the way it is specified in design files;
3) One more bugfix — it was possible to initiate an empty search and there was an error in this case, now it is fixed;
4) Lots of toolbar-related UI fixes — correct paddings and fonts;
5) Small update: I've removed all those strange extended-contact-view and contact-view-with-letter and replaced it with something more easy and convenient.

@rasom it can be checked and verified.

File: src/status_im/contacts/handlers.cljs [hunk 1]
```diff
@@ -56,6 +61,13 @@
 
 (register-handler :watch-contact (u/side-effect! watch-contact))
 
+(defn stop-watching-contact
+  [_ [_ {:keys [whisper-identity]}]]
+  (protocol/stop-watching-user! {:web3     web3
```
Comment (rasom) [inline src/status_im/contacts/handlers.cljs]:
```
WARNING: Use of undeclared Var status-im.contacts.handlers/web3 at line 66 /Users/rasom/clj/status-react/src/status_im/contacts/handlers.cljs
```

File: src/status_im/contacts/handlers.cljs [hunk 2]
```diff
@@ -31,10 +32,10 @@
 
 (defmethod nav/preload-data! :contact-list
   [db [_ _ click-handler]]
+  (dispatch [:set-in [:toolbar-search :show] nil])
```
Comment (rasom) [inline src/status_im/contacts/handlers.cljs]: Why not just use `(assoc-in db ...` here?

**Comment by alwx** [comment]
This PR became big, sorry about that.
Now it also fixes #552 and #554.

**Comment by rasom** [comment]
Ok, I will review soon, just don't add anything new please :D

**Pull request merged**